doc/src/build/prog-trans-ts-sdk.md

```diff
@@ -0,0 +1,234 @@
+
+- **For objects:** the `tx.object(objectId)` function is used to construct an input that contains an object reference.
+- **For pure values:** the `tx.pure(rawValue)` function is used, and returns an input reference that you use in commands.
+
+## Available commands
+
+The following commands are available:
+
+- `tx.splitCoin(coin, amount)` - Creates a new coin with the defined amount, split from the provided coin. Returns the coin so that it can be used in subsequent commands.
+    - Example: `tx.splitCoin(tx.gas, tx.pure(100))`
+- `tx.mergeCoins(destinationCoin, sourceCoins)` - Merges the sourceCoins into the destinationCoin.
+    - Example: `tx.mergeCoins(tx.object(coin1), [tx.object(coin2), tx.object(coin3)])`
+- `tx.transferObjects(objects, address)` - Transfers a list of objects to the specified address.
+    - Example: `tx.transferObjects([tx.object(thing1), tx.object(thing2)], tx.pure(myAddress))`
+- `tx.moveCall({ target, arguments, typeArguments  })` - Executes a move call. Returns whatever the move call returns.
+    - Example: `tx.moveCall({ target: '0x2::devnet_nft::mint', arguments: [tx.pure(name), tx.pure(description), tx.pure(image)] })`
+- `tx.makeMoveVec({ type, objects })` - Constructs a vector of objects that can be passed into a `moveCall`. This is required as there’s no way to define a vector as an input.
+    - Example: `tx.makeMoveVec({ objects: [tx.object(id1), tx.object(id2)] })`
+- `tx.publish(modules, dependencies)` - Publishes a Move package. Returns the upgrade capability object.
+
+## Passing command results as arguments
+
+You can use the result of a command as an argument in a subsequent command. Each command method on the transaction builder returns a reference to the command result.
+
+```tsx
+// Split a coin object off of the gas object:
+const coin = tx.splitCoin(tx.gas, tx.pure(100);
+// Transfer the resulting coin object:
+tx.transferObjects([coin], tx.pure(address));
+```
+
+When a command returns multiple results, you can access the result at a specific index either using destructuring, or array indexes.
+
+```tsx
+// Destructuring (preferred, as it gives you logical local names):
+const [nft1, nft2] = tx.moveCall({ target: '0x2::nft::mint_many' });
+tx.transferObjects([nft1, nft2], tx.pure(address));
+
+// Array indexes:
+const mintMany = tx.moveCall({ target: '0x2::nft::mint_many' });
+tx.transferObjects([mintMany[0], mintMany[1]], tx.pure(address));
+```
+
+## Use the gas coin
+
+With Programmable Transactions, you’re able to use the gas payment coin to construct coins with a set balance using `splitCoin`. This is useful for Sui payments, and avoids the need for up-front coin selection. You can use `tx.gas` to access the gas coin in a transaction, and is valid as input for any arguments.
```

Comment:
It can be used by-reference for any command, so merge coins and Move call in addition to split coin.